swarm/network/protocol_test.go

```diff
@@ -151,66 +99,75 @@ func (s *bzzTester) testHandshake(lhs, rhs *HandshakeMsg, disconnects ...*p2ptes
 		peers = []discover.NodeID{id}
 	}
 
-	s.TestExchanges(HandshakeMsgExchange(lhs, rhs, id)...)
-	s.TestDisconnected(disconnects...)
-}
+	err := s.TestExchanges(HandshakeMsgExchange(lhs, rhs, id)...)
```

Comment:
Should this error be returned before the possible call to s.TestDisconnected if the err is not nil?

Reply:
In there case where we expect a disconnect, we expect there to be an error returned from the handshake so it should always be non-nil anyway.

Reply:
On further testing, I was mistaken - it seems that for this handshake `TestExchanges` will only return an error if it fails in some way, but not for a rejected handshake.

I'll update the tests and rebase on `swarm-network-rewrite` again.